Mathlib/Logic/Equiv/Fin.lean

```diff
@@ -498,3 +498,13 @@ instance subsingleton_fin_zero : Subsingleton (Fin 0) :=
 /-- `Fin 1` is a subsingleton. -/
 instance subsingleton_fin_one : Subsingleton (Fin 1) :=
   finOneEquiv.subsingleton
+
+/-- The natural `Equiv` between `(Fin m → α) × (Fin n → α)` and `(Fin (m + n) → α)`.-/
+def Equiv.finArrowProdEquivFinAddArrow {α : Type*} (m n : ℕ) :
+    (Fin m → α) × (Fin n → α) ≃ (Fin (m + n) → α) where
+  toFun := fun ⟨f, g⟩ ↦ Fin.append f g
+  invFun := fun f ↦ ⟨fun i ↦ f (Fin.castAdd n i), fun i ↦ f (Fin.natAdd m i)⟩
+  left_inv := by simp [Function.LeftInverse]
+  right_inv := by
+    unfold Fin.append Fin.addCases
+    simp [Function.RightInverse, Function.LeftInverse]
```

Comment:
Slightly shorter:
```suggestion
  toFun fg := Fin.append fg.1 fg.2
  invFun f := ⟨fun i ↦ f (Fin.castAdd n i), fun i ↦ f (Fin.natAdd m i)⟩
  left_inv fg := by simp
  right_inv f := by
    unfold Fin.append Fin.addCases
    simp
```

I think the `unfold` here indicates we have a missing lemma; can you add
```lean
theorem Fin.append_castAdd_natAdd :
    (Fin.append (fun i => f (Fin.castAdd n i)) fun i => f (Fin.natAdd m i)) = f := rfl
```
near `FIn.append` so that we can avoid the `unfold`?